src/main/java/com/bravo/user/controller/PaymentController.java

```diff
@@ -0,0 +1,36 @@
+package com.bravo.user.controller;
+
+import com.bravo.user.annotation.SwaggerController;
+import com.bravo.user.dao.model.Payment;
+import com.bravo.user.dao.model.User;
+import com.bravo.user.dao.repository.PaymentRepository;
+import com.bravo.user.service.PaymentService;
+import com.bravo.user.service.UserService;
+import com.bravo.user.validator.UserValidator;
+import org.checkerframework.checker.units.qual.A;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.List;
+
+@SwaggerController
+@RequestMapping(value = "/payments")
+public class PaymentController{
+
+    @Autowired
+    private PaymentService paymentService;
+    @Autowired
+    private UserService userService;
+    @Autowired
+    private UserValidator userValidator;
+
+    @GetMapping(value = "/retrieve/{userId}/")
+    @ResponseBody
+    public List<Payment> getUserPayments(@PathVariable String userId) {
+        userValidator.validateId(userId);
+        return paymentService.getUserPayments(userId);
+    }
```

Comment:
1. Weird whitespace.
2. `GetMapping` should not end with a `/`
3. Should not return DAO model. Should use a DTO model instead.

Reply:
The DAO is Data Access Object and mirrors your database exactly. The DTO is Data Transfer Object and is sent to the clients. The DAO may have fields or logic in your database that you don't want to send to a client. You can do a conversion between the 2 types to make your API more extensible and adjustable. If your API client is relying on your database matching their contract then you will put yourself in a bad situation.

Reply:
That makes perfect sense. Thanks for that , Justin.


Overall what would you say of the over all assessment. I see I have some rookie mistakes and slightly "sloppy" code in some places, but that can easily improved on.

Reply:
Hey @Kruzik762 sorry for the delay in responding.  Bravo tries to place people where it can when it can.  It's not a pass/fail, it's more of a can we place you at x client and have their needs met? And of course do we want to work with this person :) .  Overall, it's really a question for the recruitment team.